packages/flutter/lib/src/widgets/debug.dart

```diff
@@ -114,8 +114,21 @@ bool debugPrintGlobalKeyedWidgetLifecycle = false;
 ///  * [debugProfileLayoutsEnabled], which does something similar for layout,
 ///    and [debugPrintLayouts], its console equivalent.
 ///  * [debugProfilePaintsEnabled], which does something similar for painting.
+///  * [debugProfileBuildsEnabledUserWidgets], which adds events for user-created
+///    [Widget] build times and incurs less overhead.
 bool debugProfileBuildsEnabled = false;
 
+/// Adds [Timeline] events for every user-created [Widget] built.
+///
+/// A user-created [Widget] is any [Widget] that is constructed in the root
+/// library.
```

Comment:
Can we spell this out a little more?

I'm assuming it means something like this:

```dart
import 'package:flutter/material.dart';
import 'package:foo/foo.dart'; // Exposes a `Foo` widget that creates a bunch of child widgets.

// ...
class MyWidget extends StatelessWidget {
  Widget build(BuildContext context) {
    return Container( // Framework will create some child widgets
      color: Colors.red,
      child: Foo( // Foo will create intermediate widgets between here and Container
        propA: 'Baz',
        child: Container(child: ListTile()), // ListTile creates a _bunch_ of widgets
      ),
    );
  }
}
```

Which widgets will show up when this is turned on and I do something like `runApp(MyWidget())`?

Reply:
I forgot to write what I assume: I assume the timeline would show tracing for `MyWidget`, `Container`, `Foo`, `Cotainer`, `ListTile`, and that's it?

Reply:
> I forgot to write what I assume: I assume the timeline would show tracing for MyWidget, Container, Foo, Cotainer, ListTile, and that's it?

Yep, that's correct.  What do you suggest we add now that that is clear?

Reply:
Example code that illustrates this.

Reply:
I don't know that will help because we'll be saying something like "All the widgets that aren't actually in the code aren't traced".  The interesting thing about this flag is the omission of the things that aren't present in the source code.

I think it would be better if we could explain in prose the difference between what is written now and what I just clarified to you.

How about this:
```
/// A user-created [Widget] is any [Widget] that is constructed in the root
/// library.  Often [Widget]s contain child [Widget]s that are constructed in
/// libraries (for example, a [TextButton] having a [RichText] child).  Timeline events 
/// for those children will be omitted with this flag.
```

Reply:
I think that prose makes sense. And I realize that `constructed` is meant to literally refer to `calling the constructor`. I still think it would help to clearly and explicitly call out that this will hold true whether the widget is defined in the root library or not.

Reply:
Done.